### DISCUSSION TITLE
Do we have overall/service-specific IP/user-agent bans?

### DISCUSSION THREAD

**Discussion by Moonbase59** [AzuraCast/AzuraCast#6581] [Q&A]
Reason I’m asking: I see a high amount of SFTP hacking attempts. Some of these would-be attackers hog the test streams at the same time.

I’d like to completely block access to these, if possible.

In station settings, I could ban IPs for streaming, but that would surely not stop the same IP from trying to hack my SFTP, right?

**Comment by scysys** [comment]
>In station settings, I could ban IPs for streaming, but that would surely not stop the same IP from trying to hack my SFTP, right?

Right. The options there are for the Icecast server itself.

> I see a high amount of SFTP hacking attempts.

Its normal. Mostly automated bots. If you have a good password it is no problem overall. Just ignore.

If you cant ignore it yourself, IPTables should enough if you know how to use it.
Maybe [Fail2Ban](https://www.fail2ban.org/) or a software firewall like [CSF](https://configserver.com/) is what you searching for.

**Comment by BusterNeece** [comment]
Well, we probably _could_ incorporate the IP bans into the SFTP connection side. The problem is, it would be a pretty low-yield exercise, as the IP ban would take effect at the exact same step as the auth step (after sftpgo has hit our PHP script to check if a user exists with the given credentials), so the load on the server would be comparable.

Sftpgo is, however, a ridiculously customizable piece of software, so it wouldn't surprise me if they had some sort of support for IP banlists via text file, which we could write to. It's worth looking into, and certainly worth submitting as a feature request.
